### PR TITLE
Level Progress Fixes + Floating Point Safety Function

### PR DESCRIPTION
**Pull request by Wealthyturtle**
### Description
Fixes the issue with the Level Progress expression causing an exception to occur if a NaN or INF value is passed onto the server. Additionally, this PR also adds 2 new ``safe()`` functions into the Math2 util class to make a floating point number "safe" by replacing NaN and INF values. Feedback on the documentation for those 2 functions would be good :)

---
**Target Minecraft Versions:**     Any
**Requirements:**     None
**Related Issues:**     #2680

### CODE REVIEW

File: src/main/java/ch/njol/util/Math2.java [hunk 1]
```diff
@@ -454,4 +454,30 @@ public static double smoothStep(final double x, final double x1, final double x2
...
 	
+	/**
+	 * Guarantees a float is neither NaN nor INF.
+	 * Useful for situations when safe floats are required.
+	 * 
+	 * @param f
+	 * @return 0 if f is NaN or INF, otherwise f
+	 */
+	public static float safe(float f) {
+		if(f != f || Float.isInfinite(f)) //NaN or INF
```
Comment (APickledWalrus) [inline src/main/java/ch/njol/util/Math2.java:465]: Space after `if`

File: src/main/java/ch/njol/util/Math2.java [hunk 2]
```diff
@@ -454,4 +454,30 @@ public static double smoothStep(final double x, final double x1, final double x2
...
+	
+	/**
+	 * Guarantees a double is neither NaN nor INF.
+	 * Useful for situations when safe doubles are required.
+	 * 
+	 * @param d
+	 * @return 0 if d is NaN or INF, otherwise d
+	 */
+	public static double safe(double d) {
+		if(d != d || Double.isInfinite(d)) //NaN or INF
```
Comment (APickledWalrus) [inline src/main/java/ch/njol/util/Math2.java:478]: Space after `if`

File: src/main/java/ch/njol/skript/expressions/ExprLevelProgress.java [hunk 3]
```diff
@@ -101,13 +101,13 @@ public void change(final Event e, final @Nullable Object[] delta, final ChangeMo
 					return;
 			}
 			p.setLevel(Math.max(0, p.getLevel() + (int) Math.floor(c)));
```
Comment (FranKusmiruk) [inline src/main/java/ch/njol/skript/expressions/ExprLevelProgress.java:103]: don't you have to use the method on the level as well

Comment (Wealthyturtle) [reply src/main/java/ch/njol/skript/expressions/ExprLevelProgress.java:103]: No, because it's converted to an int, which destroys the NaN/INF value.

Comment (FranKusmiruk) [reply src/main/java/ch/njol/skript/expressions/ExprLevelProgress.java:103]: Missed that bit, should be good then.

**Pull request merged**